lib/CleantalkSP/Common/Escape.php

```diff
@@ -40,17 +51,58 @@ public static function escJs($text)
      */
     public static function escTextarea($text)
     {
-        // TODO
+        return htmlspecialchars( $text, ENT_QUOTES );
     }
     
     /**
      * Simple method: escape url
      *
-     * @param $text
+     * @param $url
      */
-    public static function escUrl($text)
+    public static function escUrl($url, $protocols = null, $_context = 'display')
     {
-        // TODO
+        $original_url = $url;
+
+        if ( '' === $url ) {
+            return $url;
+        }
+
+        $url = str_replace( ' ', '%20', ltrim( $url ) );
+        $url = preg_replace( '|[^a-z0-9-~+_.?#=!&;,/:%@$\|*\'()\[\]\\x80-\\xff]|i', '', $url );
```

Comment:
Please, make sure due to a test that |regexp|i is the correct form. I haven't found any info about || usage.

Reply:
@alexandergull PHP regexp functions could use any symbols as delimiters. It's ok.